rllib/algorithms/bc/bc.py

```diff
@@ -73,3 +136,57 @@ class BC(MARWIL):
     @override(MARWIL)
     def get_default_config(cls) -> AlgorithmConfig:
         return BCConfig()
+
+    @ExperimentalAPI
+    def training_step(self) -> ResultDict:
+        if not self.config["_enable_rl_module_api"]:
+            # Using ModelV2.
+            return super().training_step()
+        else:
+            # Implement logic using RLModule and Learner API.
+            with self._timers[SAMPLE_TIMER]:
+                # Sampling from offline data.
+                if self.config.count_steps_by == "agent_steps":
+                    train_batch = synchronous_parallel_sample(
```

Comment:
Add a TODO:
* For offline Algos, we should not use RolloutWorkers (or EnvRunners) anymore.
* Instead, we should handle the (already distributed) dataset directly from within the Algorithm's `training_step`. EnvRunner/RolloutWorker should ONLY be used for actual env stepping and model inference (action computations).

Reply:
@sven1977 I have opened an issue for this #39849